src/server/index.jsx

```diff
@@ -86,6 +79,13 @@ const getBuildMetadata = () => {
   return buildMetadata;
 };
 
+const skipMiddleware = (_req, _res, next) => {
+  next();
+};
+
+const injectCspHeaderProdBuild =
+  process.env.NODE_ENV !== 'production' ? skipMiddleware : injectCspHeader;
```

Comment:
This is a work-around, to prevent `injectCspHeader` running on local development builds `npm run dev` but continue to work for production builds on all environments (local/test/live) `npm run build && npm run start).

The reason I had to prevent it from running on the dev build was that there were CSP header issues with our Canonical page webpack development JavaScript bundles that I wasn't able to resolve. Screenshots in the PR description.

Reply:
If I removed this condition, (as in, replaced with `injectCspHeader`, I'd get these following issues:

The following pages have no errors when running `npm run build && npm run start`, but do have an issue with `npm run dev`.

http://localhost:7080/pidgin - canonical front page
http://localhost:7080/pidgin/media-44221514 - canonical MAP
http://localhost:7080/pidgin/tori-51745682 - canonical STY
http://localhost:7080/pidgin/tori-49221071 - canonical PGL
http://localhost:7080/pidgin/articles/cwl08rd38l6o - canonical Optimo Article

- Given I'm running localhost `npm run dev` & I'm running `injectCspHeader` on the dev build too,
- When I visit http://localhost:7080/pidgin (any canonical page)
- Then I see CSP console errors & client-side JS is not working

```
EvalError: call to eval() blocked by CSP vendor-._node_modules_r.js:313:1
    cfHo http://localhost:1124/static/js/vendor-._node_modules_r.js:313
    __webpack_require__ http://localhost:1124/static/js/main-._src_a.js:1495
    fn http://localhost:1124/static/js/main-._src_a.js:853
    0 http://localhost:1124/static/js/main-._m.js:82
    __webpack_require__ http://localhost:1124/static/js/main-._src_a.js:1495
    checkDeferredModules http://localhost:1124/static/js/main-._src_a.js:46
    webpackJsonpCallback http://localhost:1124/static/js/main-._src_a.js:33
    <anonymous> http://localhost:1124/static/js/vendor-._node_modules_a.js:1
Content Security Policy: The page's settings blocked the loading of a resource at eval ("script-src"). vendor-._node_modules_r.js:313
```

```
vendor-._node_modules_r.js:313 Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src
```
When playing media on this MAP http://localhost:7080/afaanoromoo/23149891, with the development build `npm run dev` we see a CSP error in the console, but this is not visible with the prod build `npm run build && npm run start`
```
EvalError: call to eval() blocked by CSP vendor-._node_modules_r.js:313:1
Content Security Policy: The page's settings blocked the loading of a resource at eval ("script-src"). vendor-._node_modules_r.js:313
```
<img width="1437" alt="Screenshot 2020-04-02 at 16 47 19" src="https://user-images.githubusercontent.com/3028997/78270124-53879400-7502-11ea-928a-aa75ad226eaf.png">

If during review, anyone can suggest an alternative solution to get CSP headers working on localhost development with `npm run dev`, when the line above is replaced with `injectCspHeader` that would be appreciated.

Reply:
Hey! I'm having a review/play around with this at the moment - I'm pretty sure the `eval` error is due to source maps are being generated locally - and the call to `eval` is being blocked by the CSP. One way of fixing this is by using a different (potentially slower) method to create source maps locally: https://webpack.js.org/configuration/devtool/ - I have tried adding `devtool: 'cheap-module-source-map'`, to the client webpack config, and this seems to solve this issue.

We then get a different issue, which is that 'http://localhost:1124' and 'ws://localhost:1124' are not covered by the CSP connect-src directive, but as long as those are included in the non-live connect src directive (or maybe we add some that are just added locally?) we should be ok!

Reply:
If we would rather create a separate ticket to look into this and ignore CSP headers locally for now, I'd be happy to review this again with that in mind :)

Reply:
@tochwill Thanks for looking into this! That sounds promising. Sure, I will raise a separate issue with this, since it also impacts the webpack dev config.

Reply:
Ticket is here: https://github.com/bbc/simorgh/issues/6089 Thanks, @tochwill

Reply:
Great! Sounds like a plan to me

Reply:
I agree, I think if we have the headers available locally when we do `npm run build` that's the most useful thing.